### PR TITLE
[BAD-168] Mapr401: When using impersonation, proxy user is not picked for MapReduce operations

### PR DESCRIPTION
**Pull request by stepanovdg**
* Added list of missed resources.
* The main reason was in using local value, but not the current one, of UserGroupInformation which is set on Configuration object creation.

### CODE REVIEW

**Comment by stepanovdg** [comment]
@mattyb149, Could you please take a look?

File: mapr401/src/org/pentaho/hadoop/shim/mapr401/ConfigurationProxyV2.java [hunk 1]
```diff
@@ -38,16 +39,39 @@
...
+    // create with a null Cluster
+    JobConf jobConf = new JobConf( new org.apache.hadoop.conf.Configuration() );
+    job = new JobProxy( jobConf );
+    job.getConfiguration().addResource( "hdfs-site.xml" );
+    job.getConfiguration().addResource( "hbase-site.xml" );
+    job.getConfiguration().addResource( "core-site.xml" );
+    job.getConfiguration().addResource( "yarn-site.xml" );
+    job.getConfiguration().addResource( "mapred-site.xml" );
+    job.getConfiguration().addResource( "oozie-site.xml" );
+    job.getConfiguration().addResource( "hive-site.xml" );
```
Comment (mattyb149) [inline mapr401/src/org/pentaho/hadoop/shim/mapr401/ConfigurationProxyV2.java]: Aren't some of these (like hdfs-site.xml) added automatically in the base Configuration?

**Comment by stepanovdg** [comment]
Thank you, you are right. Core-site is added in Configuration, yarn-site and mapred-site in JobConf, hbase-site in calling hadoop shim, oozie-site and hive-site seems to be not required. 
Done.

**Comment by mattyb149** [comment]
Any way to unit test these changes?

**Comment by stepanovdg** [comment]
As this functionality is using in EE I tried to implement unit tests which would cover this functionality but really it depends on realization of the Job.submit. We need to stub half of the hadoop server side to reproduce this.

**Pull request merged**